src/generators.jl

```diff
@@ -814,3 +814,88 @@ Usage
 `A = forest_fire_graph(c, k, p)` returns a forest-fire sample. 
 `A = forest_fire_graph(A0, k, p)` 
 """
+
+
+"""
+`partial_duplication`
+=====================
+
+A random graph model based off of the evolution of protein-protein interaction
+networks. The method takes an undirected graph and runs a number of steps 
+which randomly selects an existing vertex, duplicates the node, and keeps the
+edges between its neighbor with a given probability. In the original work 
+the authors prove that new vertices will tend to become isolated for 
+p <= 0.567143. 
+
+ src: https://arxiv.org/pdf/1408.0904.pdf
+
+Input
+-----
+- 'A::MatrixNetwork{T}': seed matrix network 
+- `steps::Int': The number of steps to run the procedure
+- `new_edge_p::Float64': the probability 
+
+Output
+------
+- a new matrix network generated through the partial duplication procedure. 
+""" 
+function partial_duplication(A::MatrixNetwork{T},steps::Integer, new_edge_p::Float64) where T
+ 
+    is_undirected(A)                      || throw(ArgumentError("A must be undirected."))
+    (new_edge_p >= 0 && new_edge_p <= 1)  || throw(ArgumentError("new_edge_p must be a probability."))
+    steps >= 0                            || throw(ArgumentError("Must take a non-negative number of steps."))
+    # let it steps equal 0 for testing purposes
+
+    n,_ = size(A) # n will be updated
+
+    get_row = (csr_A,i)->(csr_A.ci[csr_A.rp[i]:csr_A.rp[i+1]-1],
```

Comment:
should probably make these external functions -- especially since you test them.

Reply:
Should this be written as a getindex function?

Reply:
no, either drop the test to them, or write it as 
`_get_row` or something in the `MatrixNetworks` namespace.